integrations/optimum/src/haystack_integrations/components/embedders/optimum_text_embedder.py

```diff
@@ -0,0 +1,200 @@
+from typing import Any, Dict, List, Optional
+
+import torch
+from haystack import component, default_from_dict, default_to_dict
+from haystack.utils import Secret, deserialize_secrets_inplace
+from haystack.utils.hf import HFModelType, check_valid_model, deserialize_hf_model_kwargs, serialize_hf_model_kwargs
+from optimum.onnxruntime import ORTModelForFeatureExtraction
+from transformers import AutoTokenizer
+
+
+class OptimumTextEmbedder:
+    """
+    A component to embed text using models loaded with the HuggingFace Optimum library.
+    This component is designed to seamlessly inference models using the high speed ONNX runtime.
+
+    Usage example:
+    ```python
+    from haystack_integrations.components.embedders import OptimumTextEmbedder
+
+    text_to_embed = "I love pizza!"
+
+    text_embedder = OptimumTextEmbedder(model="BAAI/bge-small-en-v1.5")
+    text_embedder.warm_up()
+
+    print(text_embedder.run(text_to_embed))
+
+    # {'embedding': [-0.07804739475250244, 0.1498992145061493,, ...]}
+    ```
+
+    Key Features and Compatibility:
+        - **Primary Compatibility**: Designed to work seamlessly with any embedding model present on the Hugging Face
+        Hub.
+        - **Conversion to ONNX**: The models are converted to ONNX using the HuggingFace Optimum library. This is
+        performed in real-time, during the warm-up step.
+        - **Accelerated Inference on GPU**: Supports using different execution providers such as CUDA and TensorRT, to
+        accelerate ONNX Runtime inference on GPUs.
+        Simply pass the execution provider as the onnx_execution_provider parameter. Additonal parameters can be passed
+        to the model using the model_kwargs parameter.
+        For more details refer to the HuggingFace documentation:
+        https://huggingface.co/docs/optimum/onnxruntime/usage_guides/gpu.
+    """
+
+    def __init__(
+        self,
+        model: str = "BAAI/bge-small-en-v1.5",
+        token: Optional[Secret] = Secret.from_env_var("HF_API_TOKEN", strict=False),  # noqa: B008
+        prefix: str = "",
+        suffix: str = "",
```

Comment:
Not sure what standard practice should be here, but the model that is specified by default recommends that we prepend an instruction to the query (resource [here](https://huggingface.co/BAAI/bge-small-en-v1.5#frequently-asked-questions))

> For the bge-*-v1.5, we improve its retrieval ability when not using instruction. No instruction only has a slight degradation in retrieval performance compared with using instruction. So you can generate embedding without instruction in all cases for convenience.
>
> For a retrieval task that uses short queries to find long related documents, it is recommended to add instructions for these short queries. The best method to decide whether to add instructions for queries is choosing the setting that achieves better performance on your task. In all cases, the documents/passages do not need to add the instruction.

so I wonder if we choose this model as default then we should also by default populate the `prefix` parameter?

Reply:
I think setting a default prefix is a bad idea. I have changed the default model to `sentence-transformers/all-mpnet-base-v2`, which does not require a prefix or suffix by default. This is also the default for the Sentence Transformers Embedders, so it maintains consistency.